### PR TITLE
Add "auth_method" and "auth_source" option to support explicit auth method and auth_source

### PR DESCRIPTION
**Pull request by sakama**
This PR is based on #49. So, first 6 commits are unrelated ones.
This PR will solve #50 

Currently, plugin user have to use `uri` option when they want to specify auth mechanism and auth source.
```yaml
in:
  type: mongodb
  uri: mongodb://myuser:mypassword@localhost:27017/my_database?authMechanism=SCRAM-SHA-1&authSource=another_database
```

They can't specify auth source with separated URI parameters, 
To solve this problem, I added 2 new options - `auth_method` and `auth_source`.
Plugin user will be able to set auth method and auth source explicitly.
```yaml
in:
  type: mongodb
  hosts:
  - {host: localhost, port: 27017}
  auth_method: scram-sha-1
  user: myuser
  password: mypassword
  database: my_database
  auth_source: another_database
```

MongoDB supports various kind of auth methods.
https://docs.mongodb.com/manual/core/authentication/
http://mongodb.github.io/mongo-java-driver/3.0/driver-async/reference/connecting/authenticating/

* SCRAM-SHA-1
* MONGODB-CR
* x.509
* Kerberos (GSSAPI)
* LDAP (PLAIN)

However, what I supported were only `scram-sha-1`, `mongodb-cr` and `auto`.
When plugin user choose `auto`,  mongodb-java-client will negotiate the best mechanism based on the version of the server that the client is authenticating to.
If the server version is 3.0 or higher, the driver will authenticate using the SCRAM-SHA-1 mechanism.
Otherwise, the driver will authenticate using the MONGODB_CR mechanism.

### CODE REVIEW

**Comment by sakama** [comment]
@mjalkio Related to #50, I don't have an environment that will use another auth source.
Could you try this PR if possible?
You can build by `./gradlew gem -x test` with skipping test, and able to install `gem install --local /path/to/embulk-input-mongodb-0.6.1.gem`

**Comment by mjalkio** [comment]
Hi @sakama, thank you for adding this! I won't have time to test it soon, but it looks good to me.

**Comment by sakama** [comment]
@mjalkio Got it. Thanks!

File: README.md [hunk 1]
```diff
@@ -188,44 +221,21 @@ $ ./gradlew gem
 
 ## Test
 
-```
-$ ./gradlew test  # -t to watch change of files and rebuild continuously
-```
-
-To run unit tests, we need to configure the following environment variables.
+Firstly install Docker and Docker compose then `docker-compose up -d`,
+so that an FTP server will be locally launched then you can run tests with `./gradlew test`.
```
Comment (kamatama41) [inline README.md]: It's not FTP

Comment (sakama) [reply README.md]: Thanks. Fixed 2dfbb58

**Comment by sakama** [comment]
@kamatama41 Could you take a look at this PR and https://github.com/hakobera/embulk-input-mongodb/pull/49 when you can?

**Comment by kamatama41** [comment]
Sorry for late, LGTM 👍

**Pull request merged**